src/ompl/tools/thunder/src/SPARSdb.cpp

```diff
@@ -1683,7 +1683,11 @@ void ompl::geometric::SPARSdb::connectGuards(Vertex v, Vertex vp)
     Edge e = (boost::add_edge(v, vp, g_)).first;
 
     // Add associated properties to the edge
-    edgeWeightProperty_[e] = distanceFunction(v, vp);  // TODO: use this value with astar
+    if (useCostInRoadmap_) {
+        edgeWeightProperty_[e] = costFunction(v, vp);  // TODO: use this value with astar
+    } else {
+        edgeWeightProperty_[e] = distanceFunction(v, vp);  // TODO: use this value with astar
+    }
```

Comment:
should this be an enum providing several options for weight?

Reply:
Hmm for the current use we want to feed it our own custom cost function, but I can see value in making this an enum and giving us the option to slightly deviate from the cost function we use for offline planning when using recall.
We however might never need to do so, should we wait to make this an enum or do it from now?

Reply:
I'm just saying from an API point of view you want to choose either cost or dist function as the weight. `cost = True/False` doesn't tell what the default is.